### PR TITLE
Support cross-project breakpoints copy paste

### PR DESCRIPTION
**Pull request by TrySound**
Here added utility to merge pasted breakpoints into existing ones or add new one if breakpoint does not match any.

## Code Review

- [ ] hi @kof, I need you to do
  - conceptual review (architecture, feature-correctness)
  - test it on preview
- [ ] hi @rpominov, I need you to do
  - detailed review (read every line)

## Before requesting a review

- [ ] made a self-review
- [ ] added inline comments where things may be not obvious (the "why", not "what")

## Before merging

- [ ] tested locally and on preview environment (preview dev login: 5de6)
- [ ] updated [test cases](https://github.com/webstudio-is/webstudio-builder/blob/main/apps/builder/docs/test-cases.md) document
- [ ] added tests
- [ ] if any new env variables are added, added them to `.env.example` and the `builder/env-check.js` if mandatory

### CODE REVIEW

File: apps/builder/app/shared/copy-paste/plugin-instance.ts [hunk 1]
```diff
@@ -93,11 +97,19 @@ const getTreeData = (targetInstanceSelector: InstanceSelector) => {
     treeStyleSourceIds
   );
 
-  const treeStyles = getMapValuesBy(stylesStore.get(), (styleDecl) =>
-    treeStyleSourceIds.has(styleDecl.styleSourceId)
+  const treeBreapointIds = new Set<Breakpoint["id"]>();
+  const treeStyles = getMapValuesBy(stylesStore.get(), (styleDecl) => {
+    treeBreapointIds.add(styleDecl.breakpointId);
+    return treeStyleSourceIds.has(styleDecl.styleSourceId);
+  });
```
Comment (rpominov) [inline apps/builder/app/shared/copy-paste/plugin-instance.ts:104]: Seem like we're putting into `treeStyles` only styles that satisfy `treeStyleSourceIds.has()`, but we put into `treeBreapointIds` **all** breakpoints. Is this a bug?

Comment (TrySound) [reply apps/builder/app/shared/copy-paste/plugin-instance.ts:104]: It's filter function which iterates whole map. Just didn't want to add the second iteration.

Comment (rpominov) [reply apps/builder/app/shared/copy-paste/plugin-instance.ts:104]: Yeah, but does this mean we're copying and merging all breakpoints?

Comment (kof) [reply apps/builder/app/shared/copy-paste/plugin-instance.ts:104]: we should def. only copy breakpoints that are used on the copied instances

File: apps/builder/app/shared/tree-utils.ts [hunk 2]
```diff
@@ -493,24 +495,58 @@ export const insertStyleSourceSelectionsCopyMutable = (
...
     const styleDeclCopy = {
       ...styleDecl,
       styleSourceId: newStyleSourceId,
+      breakpointId: newBreakpointId,
     };
     styles.set(getStyleDeclKey(styleDeclCopy), styleDeclCopy);
   }
 };
+
+const areBreakpointsSimilar = (left: Breakpoint, right: Breakpoint) => {
```
Comment (kof) [inline apps/builder/app/shared/tree-utils.ts:529]: this could be part of the css-engine I think, similar to matchMedia, maybe call it equalMedia()?

**Review by kof — APPROVED** [review]

File: apps/builder/app/shared/copy-paste/plugin-instance.ts (hunk 1 above)
Comment (TrySound) [reply apps/builder/app/shared/copy-paste/plugin-instance.ts:104]: You are right. I get what you mean now. Will add test

**Pull request merged**